import_obfuscation.go

```diff
@@ -0,0 +1,473 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"sort"
+	"strings"
+
+	"github.com/Binject/debug/goobj2"
+)
+
+type pkgInfo struct {
+	pkg  *goobj2.Package
+	path string
+}
+
+type dataType uint8
+
+const (
+	other dataType = iota
+	importPath
+	namedata
+)
+
+func obfuscateImports(objPath, importCfgPath string) (map[string]string, error) {
+	importCfg, err := goobj2.ParseImportCfg(importCfgPath)
+	if err != nil {
+		return nil, err
+	}
+	mainPkg, err := goobj2.Parse(objPath, "main", importCfg)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing main objfile: %v", err)
+	}
+	privatePkgs := []pkgInfo{{mainPkg, objPath}}
+
+	// build list of imported packages that are private
+	for pkgPath, info := range importCfg {
+		if isPrivate(pkgPath) {
+			pkg, err := goobj2.Parse(info.Path, pkgPath, importCfg)
+			if err != nil {
+				return nil, fmt.Errorf("error parsing objfile %s at %s: %v", pkgPath, info.Path, err)
+			}
+
+			privatePkgs = append(privatePkgs, pkgInfo{pkg, info.Path})
+		}
+	}
+
+	var sb strings.Builder
+	var buf bytes.Buffer
+	garbledImports := make(map[string]string)
+	for _, p := range privatePkgs {
+		// log.Printf("\n\n++ Obfuscating object file for %s ++\n", p.pkg.ImportPath)
+		for _, am := range p.pkg.ArchiveMembers {
+			if am.IsCompilerObj() {
+				continue
+			}
+
+			// add all private import paths to a list to garble
+			var privateImports []string
+			privateImports = append(privateImports, p.pkg.ImportPath)
+			if strings.ContainsRune(p.pkg.ImportPath, '/') {
+				privateImports = append(privateImports, importPathCombos(p.pkg.ImportPath)...)
+			}
+			for i := range am.Imports {
+				if isPrivate(am.Imports[i].Pkg) {
+					privateImports = append(privateImports, am.Imports[i].Pkg)
+					if strings.ContainsRune(am.Imports[i].Pkg, '/') {
+						privateImports = append(privateImports, importPathCombos(am.Imports[i].Pkg)...)
+					}
+					am.Imports[i].Pkg = hashImport(am.Imports[i].Pkg, garbledImports)
+				}
+			}
+			for i := range am.Packages {
+				if isPrivate(am.Packages[i]) {
+					privateImports = append(privateImports, am.Packages[i])
+					if strings.ContainsRune(am.Packages[i], '/') {
+						privateImports = append(privateImports, importPathCombos(am.Packages[i])...)
+					}
+					am.Packages[i] = hashImport(am.Packages[i], garbledImports)
+				}
+			}
```

Comment:
this 18-line chunk is almost identical to the one before with `am.Imports` - deduplicate?